torch/utils/data/datapipes/iter/grouping.py

```diff
@@ -11,7 +14,16 @@
     "UnBatcherIterDataPipe",
 ]
 
-T_co = TypeVar('T_co', covariant=True)
+T_co = TypeVar("T_co", covariant=True)
+
+def __getattr__(name: str):
+    if name in ["SHARDING_PRIORITIES", "ShardingFilterIterDataPipe"]:
+        warnings.warn(f"Please use `{name}` from the `torch.utils.data.datapipes.iter.sharding` package",
+                      category=FutureWarning, stacklevel=2)
```

Comment:
Can we add one more comment saying it's going to be removed in PyTorch 2.1?

Reply:
nit: I notice that TorchData imports from `torch.utils.data.datapipes.iter` for `ShardingFilter` and `torch.utils.data.datapipes.iter.sharding` for `SHARDING_PRIORITIES`.

Should we have separate comments for each `name`?

Reply:
> I notice that TorchData imports from torch.utils.data.datapipes.iter for ShardingFilter and torch.utils.data.datapipes.iter.sharding for SHARDING_PRIORITIES.

Good point... I think `tudd.iter.ShardingFilter` is preferred over `tudd.iter.sharding.ShardingFilterIterDataPipe`.

But if user already used `tudd.iter.grouping.ShardingFilterIterDataPipe`, perhaps the immediate rerouting would be `tudd.iter.sharding.ShardingFilterIterDataPipe` ~

Reply:
I think we should do two warnings here because users probably won't be happy to see another warning after changing the code to the proposed fix.

Reply:
Do we have a guideline on this? Since `tudd.iter.sharding.ShardingFilterIterDataPipe` is also a public API. While we encourage/promote to use `tudd.iter.ShardingFilter`, both should be supported.

If we don't want to support `tudd.iter.sharding.ShardingFilterIterDataPipe` (and should always use the `tud.iter` version, shall make it explicitly make it like  `tudd.iter.sharding._ShardingFilterIterDataPipe`? (Similar question applies to other DataPipes under `tudd.iter.XXX`?